### PR TITLE
Feature/lifemap

### PR DESCRIPTION
**Pull request by bhylak**
# What's New

- Adds a lifemap at the end of the survey
- Loads all of the indicators and the color for each
- tapping an indicator launches a dialog to say why they don't have it, what they'll do to get it, and when they'll do it
- adds to priorities in snapshot
- list on the side updates when priorities in snapshot changes (the two fragments communicate through the shared view model)
- tapping on an existing priority lets you edit the contents 
- only green indicators can be selected.

# Known Bugs

- Popup dialog not resizing for keyboard. Not sure if this is an issue with my layout or the popup dialog class
- 5 Priority Max not enforced.

### CODE REVIEW

**Comment by bhylak** [comment]
sorry about the number of files.. did a little bit of class/resource renaming.

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LifeMapFragment.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                String title = response.getIndicator().getTitle();
+                mTitle.setText(title);
+
+                mPopupWindow = new PriorityDetailPopupWindow.Builder(itemView.getContext()).
+                        setIndicatorOption(response).setResponseCallback(mCallback).build();
+
+                itemView.setOnClickListener((view)->{
+                    if(mResponse.getLevel()== IndicatorOption.Level.Green)
+                    {
+                        Toast.makeText(view.getContext(), "Choose a Red or Yellow Indicator", Toast.LENGTH_SHORT).show();
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/LifeMapFragment.java]: Needs a spanish resource

File: app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyChoosePrioritiesFragment.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * Top most fragment for displaying the life map
+ */
+
+public class SurveyChoosePrioritiesFragment extends AbstractSurveyFragment {
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setShowFooter(false);
+        setTitle("Life Map");
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/fragments/SurveyChoosePrioritiesFragment.java]: Needs a spanish resourse

File: app/src/main/java/org/fundacionparaguaya/advisorapp/viewcomponents/PriorityDetailPopupWindow.java [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+    public static class Builder extends BlurPopupWindow.Builder<PriorityDetailPopupWindow> {
+
+        LifeMapPriority mPriority = null;
+        IndicatorOption mIndicatorOption = null;
+        PriorityPopupResponseCallback mCallback;
+
+        public Builder(Context context) {
+            super(context);
+            this.setScaleRatio(0.25f).setGravity(Gravity.CENTER).setBlurRadius(10).setTintColor(0x30000000).setDismissOnTouchBackground(false);
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/viewcomponents/PriorityDetailPopupWindow.java]: Can these be set in the layout XML?

File: app/src/main/java/org/fundacionparaguaya/advisorapp/viewmodels/SharedSurveyViewModel.java [hunk 4]
```diff
@@ -341,7 +383,7 @@ protected Void doInBackground (Void ... voids) {
         @Override
         protected void onPostExecute(Void aVoid) {
             super.onPostExecute(aVoid);
-            viewModelReference.get().setSurveyState(SurveyState.COMPLETE);
+           // viewModelReference.get().setSurveyState(SurveyState.COMPLETE);
```
Comment (krconv) [inline app/src/main/java/org/fundacionparaguaya/advisorapp/viewmodels/SharedSurveyViewModel.java]: Why did this get commented out?

File: app/src/main/res/layout/fragment_prioritylist.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    <Button android:layout_width="wrap_content" android:layout_height="40dp"
+            android:id="@+id/btn_prioritylist_save"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:layout_marginBottom="24dp"
+
+            android:textAppearance="@style/CardHeading.Light"
+            android:background="@drawable/all_button_large"
+            android:text="Save"
```
Comment (krconv) [inline app/src/main/res/layout/fragment_prioritylist.xml]: Needs a spanish resource

File: app/src/main/res/values-es/strings.xml [hunk 6]
```diff
@@ -97,6 +97,14 @@
     <string name="surveyactivity_discard_snapshot">Descartar Instantánea</string>
     <string name="your_response">Tu Respuesta</string>
 
+    <string name="prioritypopup_strategyquestion">What will you do to get it?</string>
```
Comment (krconv) [inline app/src/main/res/values-es/strings.xml]: Spanish translation for all of these

File: app/build.gradle [hunk 7]
```diff
@@ -4,18 +4,23 @@ android {
...
+    configurations.all {
+        exclude group: 'com.android.support', module: 'support-v4'
+        resolutionStrategy.force 'com.google.code.findbugs:jsr305:2.0.1'
     }
 
     defaultConfig {
         applicationId "org.fundacionparaguaya.advisorapp"
         minSdkVersion 19
         targetSdkVersion 27
+        multiDexEnabled true
```
Comment (krconv) [inline app/build.gradle]: What is this? Does it support API 19? I'm getting an error when trying to launch

File: app/src/main/res/layout/fragment_prioritylist.xml (hunk 5 above)
Comment (bhylak) [reply app/src/main/res/layout/fragment_prioritylist.xml]: gah.

File: app/src/main/res/values-es/strings.xml (hunk 6 above)
Comment (bhylak) [reply app/src/main/res/values-es/strings.xml]: gah

File: app/src/main/java/org/fundacionparaguaya/advisorapp/viewcomponents/PriorityDetailPopupWindow.java (hunk 3 above)
Comment (bhylak) [reply app/src/main/java/org/fundacionparaguaya/advisorapp/viewcomponents/PriorityDetailPopupWindow.java]: nope.

**Pull request merged**